app/assets/javascripts/pretty-text/addon/upload-short-url.js

```diff
@@ -199,22 +199,19 @@ const SHORT_URL_ATTRIBUTES =
   "img[data-orig-src], a[data-orig-href], source[data-orig-src]";
 
 export function resolveCachedShortUrls(siteSettings, scope, opts) {
-  let shortUploadElements = scope.querySelectorAll(SHORT_URL_ATTRIBUTES);
+  const shortUploadElements = scope.querySelectorAll(SHORT_URL_ATTRIBUTES);
 
   if (shortUploadElements.length > 0) {
     _loadCachedShortUrls(shortUploadElements, siteSettings, opts);
   }
 }
 
 export function resolveAllShortUrls(ajax, siteSettings, scope, opts) {
-  let shortUploadElements = scope.querySelectorAll(SHORT_URL_ATTRIBUTES);
+  resolveCachedShortUrls(siteSettings, scope, opts);
```

Comment:
Let's make it clear we're first checking all the cached URLs before loading the rest.